### PR TITLE
New layer dialog improvements

### PR DESCRIPTION
**Pull request by nyalldawson**
Two different things:
- some follow up improvements on the new layer dialogs (@nirvn)
- a fix to allow the ogr provider to correctly create layers with M values (@rouault )

### CODE REVIEW

**Comment by nyalldawson** [comment]
@rouault is it a QGIS bug, or file format limitation that creating a pointz/linestringz/polygonz shapefile results in a pointzm/linestringzm/polygonzm layer?

File: src/gui/qgsnewgeopackagelayerdialog.cpp [hunk 1]
```diff
@@ -70,18 +70,18 @@ QgsNewGeoPackageLayerDialog::QgsNewGeoPackageLayerDialog( QWidget *parent, Qt::W
...
-  mGeometryTypeBox->addItem( tr( "Multi polygon" ), wkbMultiPolygon );
+  mGeometryTypeBox->addItem( tr( "MultiPoint" ), wkbMultiPoint );
+  mGeometryTypeBox->addItem( tr( "MultiLine" ), wkbMultiLineString );
+  mGeometryTypeBox->addItem( tr( "MultiPolygon" ), wkbMultiPolygon );
 
 #if 0
   // QGIS always create CompoundCurve and there's no real interest of having just CircularString. CompoundCurve are more useful
   mGeometryTypeBox->addItem( tr( "Circular string" ), wkbCircularString );
 #endif
   mGeometryTypeBox->addItem( tr( "Compound curve" ), wkbCompoundCurve );
```
Comment (nirvn) [inline src/gui/qgsnewgeopackagelayerdialog.cpp]: Should this be CompoundCurve instead? (same for CurvePolygon, CircularString, etc.)

File: src/ui/qgsnewgeopackagelayerdialogbase.ui [hunk 2]
```diff
@@ -478,28 +478,62 @@
...
   <tabstop>mRemoveAttributeButton</tabstop>
+  <tabstop>mLayerIdentifierEdit</tabstop>
+  <tabstop>mLayerDescriptionEdit</tabstop>
+  <tabstop>mFeatureIdColumnEdit</tabstop>
+  <tabstop>mGeometryColumnEdit</tabstop>
+  <tabstop>mCheckBoxCreateSpatialIndex</tabstop>
  </tabstops>
  <resources>
   <include location="../../images/images.qrc"/>
+  <include location="../../images/images.qrc"/>
```
Comment (nirvn) [inline src/ui/qgsnewgeopackagelayerdialogbase.ui]: What's happening here, a few too many? 😄

File: src/ui/qgsnewvectorlayerdialogbase.ui [hunk 3]
```diff
@@ -347,7 +356,36 @@
   <tabstop>mAttributeView</tabstop>
   <tabstop>mRemoveAttributeButton</tabstop>
  </tabstops>
- <resources/>
+ <resources>
+  <include location="../../images/images.qrc"/>
+  <include location="../../images/images.qrc"/>
```
Comment (nirvn) [inline src/ui/qgsnewvectorlayerdialogbase.ui]: Same here, a few too many.

**Review by nirvn — COMMENTED** [review]
@nyalldawson , great. A few comments added.

File: src/ui/qgsnewgeopackagelayerdialogbase.ui (hunk 2 above)
Comment (nyalldawson) [reply src/ui/qgsnewgeopackagelayerdialogbase.ui]: It's a bug in qt creator. They are everywhere now, so I figure we just wait to qt fixes it.

**Pull request merged**